woocommerce/woocommerce-ios
# [Product Bundles] Add product bundle properties to Product model

Part of: #8953


## Description


This PR adds properties specific to Product Bundles (a custom product type) to the `Product` model. (This is not all of the model changes required. Another PR will add an array of bundled items to the `Product` model, including a new model for those items.)

Apologies for the size of the PR. Many of the changes are from a new JSON mock and adding the properties to the generated fakeable and copiable methods, mocks, etc.

These are the new properties:

* `bundleLayout`
   * Also adds a new type `ProductBundleLayout` for this property. Includes core options and a custom option in case the value is modified by another extension.
* `bundleFormLocation`
   * Also adds a new type `ProductBundleFormLocation` for this property. Includes core options and a custom option in case the value is modified by another extension.
* `bundleItemGrouping`
   * Also adds a new type `ProductBundleItemGrouping` for this property. Includes core options and a custom option in case the value is modified by another extension.
* `bundleMinSize`
   * This is expected to be an integer. If the property is not set the API returns an empty string, in which case we set the value to `nil`.
* `bundleMaxSize`
   * This is expected to be an integer. If the property is not set the API returns an empty string, in which case we set the value to `nil`.
* `bundleEditableInCart`
* `bundleSoldIndividuallyContext`
   * Also adds a new type `ProductBundleSoldIndividuallyContext` for this property. This only returns two possible values (not modifiable) so it is a simple `enum`.
* `bundleStockStatus`
   * Bundle stock statuses are the same as product stock statuses, with the addition of an "Insufficient stock" status. This new status is added to `ProductStockStatus` to support all possible statuses.
* `bundleStockQuantity`
   * This is expected to be an integer. If the property is not set the API returns an empty string, in which case we set the value to `nil`.

We will use these properties to display them (read-only for now) in the product details for product bundles.

## Testing instructions


These new properties aren't yet used in the app, but you can confirm the product list loads as expected:

1. Build and run the app.
2. Open the Products tab and confirm your products load.
3. Select a product and confirm the product details load.


---
